src/cr/cube/cube_slice.py

```diff
@@ -336,3 +338,48 @@ def get_shape(self, prune=False):
         # Eliminate dimensions that get reduced to 1
         # (e.g. single element categoricals)
         return tuple(n for n in shape if n > 1)
+
+    def _prepare_index_base(self, axis):
+        # First get the margin of the opposite direction of the index axis.
+        # We need this in order to end up with the right shape of the
+        # numerator vs denominator.
+        base = self.margin(axis=(1 - axis), include_missing=True)
+
+        # Now check if the shape of the marginal needs to be fixed, because
+        # different versions of the MR containing cubes, combined with
+        # different margin directions, provide marginals of different shapes.
+        # We also need to calculate the percentage marginals correctly,
+        # so we need to perform the addition (to get the denominator)
+        # across the correct axis.
+        if axis == self.mr_dim_ind:
+            base = base / np.sum(base, axis=1)[:, None]
+            return base[:, 0]
+        elif isinstance(self.mr_dim_ind, tuple) and axis in self.mr_dim_ind:
+            total = np.sum(base, axis=(axis + 1))[0]
+            if axis == 0:
+                return base[:, 0] / total[0]
+            return base[0, :, 0] / total
```

Comment:
Gotcha!!! The `CubeSlice`, where all of this is implemented, will only ever have the max of 2 dims. So, this will work, but we might want to add a test for it. LMK if you have a good one, happy to implement it. We might want to explicitly throw an exception, similar to what you did in R. I can do that if you think it's useful, shall I?